### PR TITLE
automatically perform swap estimation on user input

### PR DESCRIPTION
**Pull request by TalDerei**
references https://github.com/penumbra-zone/web/issues/1919

https://github.com/user-attachments/assets/bd5707f2-df2e-4929-971f-ea36d6650c7b

### CODE REVIEW

File: apps/minifront/src/components/swap/swap-form/simulate-swap-result/traces/index.tsx [hunk 1]
```diff
@@ -29,17 +29,14 @@ export const Traces = ({
...
-              <ValueViewComponent view={input} size='sm' />
-              <ArrowDown size={17} className='relative z-10' />
-            </div>
-            <div className='flex flex-col items-end gap-2'>
-              <ValueViewComponent view={output} size='sm' />
-              <ArrowUp size={17} className='relative z-10' />
+          <div className='relative flex items-center justify-between'>
+            <ValueViewComponent view={input} size='sm' />
+            <div className='absolute left-1/2 -translate-x-1/2 flex items-center'>
+              <ArrowRight size={20} strokeWidth={2.5} className='text-white' />
```
Comment (TalDerei) [inline apps/minifront/src/components/swap/swap-form/simulate-swap-result/traces/index.tsx:35]: thoughts about replacing the up and down arrows with a single right arrow? the up and down arrows had alignment issues as the routes expanded

-------------

_before_

Screenshot 2024-11-21 at 7 49 07 PM

-------------

_after_

Screenshot 2024-11-21 at 7 49 33 PM

File: apps/minifront/src/components/swap/swap-form/simulate-swap.tsx [hunk 2]
```diff
@@ -3,16 +3,33 @@ import { SimulateSwapResult } from './simulate-swap-result';
...
 export const SimulateSwap = ({ layoutId }: { layoutId: string }) => {
-  const { simulateSwap, disabled, result } = useStoreShallow(simulateSwapSelector);
+  const { simulateSwap, resetSimulateSwap, disabled, result, amount } =
+    useStoreShallow(simulateSwapSelector);
+
+  useEffect(() => {
+    if (!disabled && amount && parseFloat(amount) > 0) {
+      // Prevents re-triggering the swap simulation if it's already computed
+      if (!result) {
+        void simulateSwap();
```
Comment (TalDerei) [inline apps/minifront/src/components/swap/swap-form/simulate-swap.tsx:26]: comment: we can also wrap this in a framer motion div for smoother animations

Comment (hdevalence) [reply apps/minifront/src/components/swap/swap-form/simulate-swap.tsx:26]: Could be future work

**Review by hdevalence — COMMENTED** [review]
Looks great! 🚀

**Review by hdevalence — APPROVED** [review]

File: apps/minifront/src/components/swap/swap-form/simulate-swap-result/traces/trace/index.tsx [hunk 3]
```diff
@@ -27,9 +27,12 @@ export const Trace = ({
...
 }) => {
   return (
-    <div className='flex w-full items-center justify-between gap-8'>
+    <div className='flex w-full justify-between gap-8'>
       {trace.value.map((value, index) => (
-        <div key={index} className='flex shrink-0 flex-col gap-1'>
+        <div
+          key={index}
+          className='flex min-w-[60px] max-w-[150px] shrink-0 grow basis-auto flex-col items-center gap-1'
+        >
```
Comment (TalDerei) [inline apps/minifront/src/components/swap/swap-form/simulate-swap-result/traces/trace/index.tsx:35]: tightening pill to dynamically shrink and fit within the available space

--------------------

_before_

Screenshot 2024-11-21 at 10 48 58 PM


--------------

_after_

Swaps are filed in descending order of price to get you the best possible trade for your tokens

File: apps/minifront/src/components/swap/swap-form/simulate-swap.tsx [hunk 4]
```diff
@@ -3,16 +3,33 @@ import { SimulateSwapResult } from './simulate-swap-result';
...
+  amount: state.swap.amount,
 });
 
+// Automatically trigger swap simulation on user input
 export const SimulateSwap = ({ layoutId }: { layoutId: string }) => {
-  const { simulateSwap, disabled, result } = useStoreShallow(simulateSwapSelector);
+  const { simulateSwap, resetSimulateSwap, disabled, result, amount } =
+    useStoreShallow(simulateSwapSelector);
+
+  useEffect(() => {
```
Comment (grod220) [inline apps/minifront/src/components/swap/swap-form/simulate-swap.tsx:22]: Not critical to fix, but in the future, we should always see if it's possible to use react-query first for all async tasks. The complexities of loading/error/cache invalidation is abstracted away.

Comment (TalDerei) [reply apps/minifront/src/components/swap/swap-form/simulate-swap.tsx:22]: totally, sorry about that – that's an easy refactor

**Pull request merged**